### PR TITLE
Set windows focus to strict in GNOME

### PR DESCRIPTION
**Pull request by mloviska**
Default setting of windows focus is *smart*, which can lead to focus
stealing mainly when a test starts a program right after system start
up as it is in [updates_packagekit_gpk](https://openqa.opensuse.org/tests/2489474#step/updates_packagekit_gpk/31).

- ticket:  [[leap][qe-core] test fails in xterm - mistyping test string due to lost focus](https://progress.opensuse.org/issues/111752)
- VRs:
  - [gnome@64bit-2G](http://kepler.suse.cz/tests/18232#step/updates_packagekit_gpk/33) - [opensuse-15.4-DVD-Updates-x86_64-Build20220729-5-gnome@64bit-2G](http://kepler.suse.cz/tests/18281#step/updates_packagekit_gpk/33)
  - [upgrade_Leap_15.2_gnome@64bit](http://kepler.suse.cz/tests/18182#step/system_prepare/16) - [upgrade_Leap_15.2_gnome@64bit](http://kepler.suse.cz/tests/18230#step/updates_packagekit_gpk/36)
  - [full test suite](http://kepler.suse.cz/tests/18332#)

### CODE REVIEW

**Review by foursixnine — APPROVED** [review]

**Comment by a-kpappas** [comment]
Could you add POD documentation?

**Comment by foursixnine** [comment]
@a-kpappas any extra comments before merging?

**Comment by mloviska** [comment]
So guys, are you OK merging this PR?

**Comment by Vogtinator** [comment]
Sounds like a product bug to me - if there's only a single window, there's no other option than to focus that, isn't there?

**Comment by mloviska** [comment]
> Sounds like a product bug to me - if there's only a single window, there's no other option than to focus that, isn't there?

I agree, as it is specific to GNOME, but I have not reproduced it manually. Thus, I do not really have any real lead over here. As it happens after reboot, might there be some start up program that might steal the focus?

I have also tried to add a call sleep in `updates_packagekit_gpk::setup_system()` which helped as well. 
Any suggestions how to debug it more thoroughly are very much appreciated.

**Comment by Vogtinator** [comment]
> I have also tried to add a call sleep in updates_packagekit_gpk::setup_system() which helped as well.
Any suggestions how to debug it more thoroughly are very much appreciated.

I'd say file a product bug and ask the GNOME maintainers for some ideas?

**Comment by mloviska** [comment]
> > I have also tried to add a call sleep in updates_packagekit_gpk::setup_system() which helped as well.
> > Any suggestions how to debug it more thoroughly are very much appreciated.
> 
> I'd say file a product bug and ask the GNOME maintainers for some ideas?

I had a small chat with Thomas regarding this issue as of now. He was more keen on changing the settings. I will open a bug and let's see. Thanks

**Review by a-kpappas — APPROVED** [review]

**Comment by a-kpappas** [comment]
> Sounds like a product bug to me - if there's only a single window, there's no other option than to focus that, isn't there?

I am not a GNOME user, but the desktop could be focused. 

> I have also tried to add a call sleep in `updates_packagekit_gpk::setup_system()` which helped as well. Any suggestions how to debug it more thoroughly are very much appreciated.

Press Ctrl+A on a desktop with folders.

I think that testing the focusing behaviour is beyond our scope so LGTM

**Comment by a-kpappas** [comment]
> > > I have also tried to add a call sleep in updates_packagekit_gpk::setup_system() which helped as well.
> > > Any suggestions how to debug it more thoroughly are very much appreciated.
> > 
> > 
> > I'd say file a product bug and ask the GNOME maintainers for some ideas?
> 
> I had a small chat with Thomas regarding this issue as of now. He was more keen on changing the settings. I will open a bug and let's see. Thanks

Perhaps add the change as a softfail.

**Comment by mloviska** [comment]
breaks: http://kepler.suse.cz/tests/18413#step/sshxterm/6 reverting

**Pull request merged**
